packages/pwa-kit-extension-sdk/src/react/utils/helpers.ts

```diff
@@ -23,3 +23,32 @@ export const applyHOCs = <T extends React.ComponentType<any>>(Component: T, hocs
         return hoistNonReactStatics(WrappedComponent, AccumulatedComponent) as T
     }, Component)
 }
+
+/**
+ * Applies a cache to a method on an instance.
+ * 
+ * @param instance - The instance on which to apply the cache.
+ * @param methodName - The name of the method to cache.
+ * @param cacheProperty - The name of the property to use for caching.
+ */
+export function applyCacheForMethod(instance: any, methodName: string, cacheProperty: string) {
+    const originalMethod = instance[methodName]
+
+    if (typeof originalMethod === 'function') {
+        instance[methodName] = function (this: any, ...args: any[]) {
```

Comment:
Are we changing the method signature here? I see the first argument is `this`, which is not there in the original method.

I think what you probably wanted is to use [`bind`](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Function/bind):
`instance[methodName] = function (...args: any[]) { }.bind(instance)`

Reply:
ah thanks for catching! Looks like i missed that when I moved this from a private method to util